server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/LocalResources.java

```diff
@@ -78,18 +80,35 @@ boolean isLocalEmail(MailAddress mailAddress) {
             if (!isLocalServer(mailAddress.getDomain())) {
                 return false;
             }
-            try {
-                return isLocaluser(mailAddress)
-                    || isLocalAlias(mailAddress);
-            } catch (UsersRepositoryException e) {
-                throw new RuntimeException("Unable to retrieve users", e);
-            } catch (RecipientRewriteTable.ErrorMappingException | RecipientRewriteTableException e) {
-                throw new RuntimeException("Unable to retrieve RRTs", e);
-            }
+            return belongsToALocalUser(mailAddress);
         }
         return false;
     }
 
+    private boolean belongsToALocalUser(MailAddress mailAddress) {
+        try {
+            return isLocaluser(mailAddress)
+                || isLocalAlias(mailAddress);
+        } catch (UsersRepositoryException e) {
+            throw new RuntimeException("Unable to retrieve users", e);
+        } catch (RecipientRewriteTable.ErrorMappingException | RecipientRewriteTableException e) {
+            throw new RuntimeException("Unable to retrieve RRTs", e);
+        }
+    }
+
+    Collection<MailAddress> localEmails(Collection<MailAddress> mailAddresses) {
+        return mailAddresses.stream()
```

Comment:
is it different from https://github.com/linagora/james-project/pull/3285/commits/9ca03f00a458e2fb240eb195e9686a3ab1eadece#diff-1d5da02fef94f1857012277642513ed5R40 ?

Reply:
Yes. It does domain check and user check while the one you are pointing just do domain check.

Reply:
could you do similar refactoring as suggested for the other one?